searchcore/src/vespa/searchcore/proton/matching/query.cpp

```diff
@@ -40,7 +40,25 @@ using std::vector;
 namespace proton::matching {
 
 namespace {
-void AddLocationNode(const string &location_str, Node::UP &query_tree, Location &fef_location) {
+
+Node::UP
+inject(Node::UP query, Node::UP to_inject) {
+    if (dynamic_cast<search::query::And *>(query.get())) {
```

Comment:
you could save a dynamic cast here by using the form:

```
if (auto *my_and = dynamic_cast<search::query::And *>(query.get())) {
    my_and->append(std::move(to_inject));
} ...
```

Reply:
Oooh, something new.....

Reply:
Fixed